e2e/testcases/admission_test.go

```diff
@@ -141,55 +148,84 @@ metadata:
 }
 
 func TestDisableWebhookConfigurationUpdate(t *testing.T) {
-	webhook := []byte(`
-apiVersion: admissionregistration.k8s.io/v1
-kind: ValidatingWebhookConfiguration
-metadata:
-  annotations:
-    configsync.gke.io/webhook-configuration-update: disabled
-  name: admission-webhook.configsync.gke.io
-  labels:
-    configmanagement.gke.io/system: "true"
-    configmanagement.gke.io/arch: "csmr"
-`)
-
 	nt := nomostest.New(t, nomostesting.DriftControl)
 
+	// Test starts with Admission Webhook already installed
+	nomostest.WaitForWebhookReadiness(nt)
+
 	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/hello/ns.yaml", fake.NamespaceObject("hello")))
 	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add Namespace"))
 	if err := nt.WatchForAllSyncs(); err != nil {
 		nt.T.Fatal(err)
 	}
 
-	if err := os.WriteFile(filepath.Join(nt.TmpDir, "webhook.yaml"), webhook, 0644); err != nil {
-		nt.T.Fatalf("failed to create a tmp file %v", err)
+	helloNS := &corev1.Namespace{}
+	if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
+		nt.T.Fatal(err)
 	}
+	version := helloNS.ResourceVersion
 
-	// Recreate the admission webhook
-	if _, err := nt.Shell.Kubectl("replace", "-f", filepath.Join(nt.TmpDir, "webhook.yaml")); err != nil {
-		nt.T.Fatalf("failed to replace the admission webhook %v", err)
-	}
+	nomostest.StopWebhook(nt)
 
-	// Verify that the webhook is disabled.
-	if _, err := nt.Shell.Kubectl("delete", "ns", "hello"); err != nil {
-		nt.T.Fatalf("failed to run `kubectl delete ns hello` %v", err)
+	tg := taskgroup.New()
+	tg.Go(func() error {
+		return nt.Validate(
```

Comment:
The reconciler Deployment may not be reconciled immediately. A direct validation may result in test flakiness. You may want to watch for the update.

Reply:
Would watchForCurrentStatus work for this check? I'm using it just to give enough head room for the reconciliation, similar to checking the resource version change in the test namespace - only check for the annotation _after_ the version is updated.

Reply:
> Would watchForCurrentStatus work for this check?

`watchForCurrentStatus` doesn't work because the reconciler Deployment may not be updated yet. Also it doesn't allow you to pass in a predicate.

> I'm using it just to give enough head room for the reconciliation

Even though the new reconciler Deployment is current, it is not guaranteed that the parse-apply-watch loop has completed. And `nt.WatchForAllSyncs` doesn't help either because the commit SHA is the same. 

> similar to checking the resource version change in the test namespace - only check for the annotation after the version is updated.

The resource version can be changed by various reasons. It is possible that the resource version changed, but the version has not been updated yet. I would suggest watch for the version update directly.

Reply:
So for the reconciler Deployment, you can actually have

```
tg.Go(func() error {
		predicates := []testpredicates.Predicate{
			testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus),
			testpredicates.DeploymentMissingEnvVar(reconcilermanager.Reconciler, reconcilermanager.WebhookEnabled),
		}
		return nt.Watcher.WatchObject(kinds.Deployment(),
			core.RootReconcilerName(configsync.RootSyncName), configsync.ControllerNamespace, predicates)
	})
```

For the test Namespace, you can have 
```
tg.Go(func() error {
		return nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
			[]testpredicates.Predicate{
				testpredicates.MissingAnnotation(metadata.DeclaredFieldsKey),
			})
	})
```

Reply:
Done! Thanks for the explanation!